pkg/printers/internalversion/printers.go

```diff
@@ -793,36 +791,36 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 			container := pod.Status.ContainerStatuses[i]
 
 			restarts += int(container.RestartCount)
-			if container.State.Waiting != nil && container.State.Waiting.Reason != "" {
+			switch {
+			case container.State.Waiting != nil && container.State.Waiting.Reason != "":
 				reason = container.State.Waiting.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
+			case container.State.Terminated != nil && container.State.Terminated.Reason != "":
 				reason = container.State.Terminated.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason == "" {
+			case container.State.Terminated != nil && container.State.Terminated.Reason == "":
```

Comment:
```go
case container.State.Terminated != nil:
    if container.State.Terminated.Reason != "" {
        reason = container.State.Terminated.Reason
    } else {
        reason = fmt.Sprintf("ExitCode:%d", container.State.Terminated.ExitCode)
    }
    ....

Reply:
I don't think that's the right thing to do，We should avoid using else to avoid nesting too deeply

Reply:
having all the branching be done in the `case` statements is easier to read